Desktop/analysis/analysisform.cpp

```diff
@@ -683,10 +686,25 @@ void AnalysisForm::blockValueChangeSignal(bool block, bool notifyOnceUnblocked)
 	else
 	{
 		_signalValueChangedBlocked--;
-		if (_signalValueChangedBlocked < 0)	_signalValueChangedBlocked = 0;
+		
+		if (_signalValueChangedBlocked < 0)	
+			_signalValueChangedBlocked = 0;
 
-		if (_signalValueChangedBlocked == 0 && notifyOnceUnblocked && _analysis)
-			_analysis->boundValueChangedHandler();
+		if (_signalValueChangedBlocked == 0)
+		{
+			if(notifyOnceUnblocked && _analysis)
+				_analysis->boundValueChangedHandler();
+		
+			if(_analysis->wasUpgraded()) //Maybe something was upgraded and we want to run the dropped rscripts (for instance for https://github.com/jasp-stats/INTERNAL-jasp/issues/1399)
```

Comment:
If it is not upgrading, and the notifyOnceUnblocked is true, the R scripts must be run also.

Reply:
Ok, but if the rscripts run they might change the boundvalue and cause it to run the analysis on return of script?

Reply:
But here you are in the situation that the valueChanged signal was blocked (and the R script was not sent but cached), and that now the valueChanged signal is unblocked: if this function is called with notifyOnceUnblocked true, then you expect normally that it runs the R scripts. This is a case that does not occur in JASP now (this notifyOnceUnblocked is false when setting up analyses from a JASP file), but in the future this function could be used in another way, and you would expect these R scripts to be run.

Reply:
Ok, I've fixed this